src/motor-operations.cc

```diff
@@ -84,19 +85,97 @@ static char test_acceleration_ok(float acceleration) {
 }
 #endif
 
+struct history_slot{
+    int steps[BEAGLEG_NUM_MOTORS];
+    int max_axis_steps;
+};
+
+typedef RingQueue<struct history_slot, QUEUE_LEN> SlotHistory;
+typedef FixedArray<unsigned,BEAGLEG_NUM_MOTORS> SlotPartial;
+
+
+// This class is delegated to handle the update of the absolute
+// position in steps of each motor and it should be callbacked whenever this operations
+// has to be performed.
+class MotionQueueMotorOperations::StatusListener : public MotionQueue::EventReceiver {
+public:
+    StatusListener(MotionQueue *backend, MotorCoordSystem *current_position_steps)
+                        : backend_(backend), current_position_steps_(current_position_steps) {};
```

Comment:
Try to stay within 80 characters.
There is an unnecessary semicolon at the end of that line.